libfreerdp/codec/progressive.c

```diff
@@ -361,18 +362,80 @@ static INLINE BOOL progressive_set_surface_data(PROGRESSIVE_CONTEXT* progressive
 	return TRUE;
 }
 
-static INLINE void* progressive_get_surface_data(PROGRESSIVE_CONTEXT* progressive, UINT16 surfaceId)
+static INLINE PROGRESSIVE_SURFACE_CONTEXT*
+progressive_get_surface_data(PROGRESSIVE_CONTEXT* progressive, UINT16 surfaceId)
 {
-	ULONG_PTR key;
+	void* key = (void*)(((ULONG_PTR)surfaceId) + 1);
 	void* pData = NULL;
```

Comment:
Would be nice to index the `progressive->SurfaceContexts` hashtable with the address of `PROGRESSIVE_SURFACE_CONTEXT->id` like we did in https://github.com/FreeRDP/FreeRDP/blob/master/channels/geometry/client/geometry_main.c#L484 and https://github.com/FreeRDP/FreeRDP/blob/master/channels/geometry/client/geometry_main.c#L229 . That would allow us to remove the `+1` hack for the surfaceId.